### ISSUE TITLE
Usability issue with `Sync` constraint requirement.

### ISSUE THREAD

**Issue by hejfelix** [typelevel/log4cats#492]
The fact that all the constructor methods in `Slf4jLogger` require a `Sync` instance is very detrimental to the usage of this library -- especially when considering that the macro call needs to live inside every class to produce the desired class name in the log lines. This effectively means all classes that log need the `Sync` constraint. 
I'm not sure there is an easy solution here, though :/

**Comment by FunFunFine** [comment]
There is a solution, we can either incapsulate Logger creation ability or delay ability. 

The first option is kind of obvious, here is the simplified example:
 ```scala
trait Slf4jLoggers[F[_]] {
  def make: Slf4jLogger[F]
}
object Slf4jLoggers {
  implicit def fromSync[F[_]: Sync]: Slf4jLoggers[F] = //... delay unsafe loggers creation
}

class MyService[F[_]: Slf4jLoggers] {
  val logger: Logger[F] = Slf4jLogger.getForClassOrSomethingElse
}
```
But this would require us to rework existing code, make some tough design decisions and is not what I would call an easy solution.

The second option can be seen in some libraries already, e.g. tofu: [it's the Delay typeclass,](https://github.com/tofu-tf/tofu/blob/e3551e30f789154e9154ef78f87372c4aeb1d1f9/modules/kernel/src/main/scala/tofu/Delay.scala) which is "weaker" than Sync. Its instance can be derived directly from the Sync instance, but not the other way around.
Then the logger creation would require only `Delay` in scope, so the code would be less detrimental.

@danicheg is it possible to use `tofu.Delay` here or even simply copy-paste it into the project (just like with the laconic syntax)? 
Or perhaps the factory option looks more pleasant, WDYT?

**Comment by FunFunFine** [comment]
I believe this issue can be closed as the #629 was released.
